source/js/timezone-fixers/tiktok.js

```diff
@@ -12,45 +12,59 @@ const attachedElements = new Set();
  * 	  HARD TODO: is it possible to do it for the "you may like" section on the right side of a video page? there is no id in the html but perhaps the click event listener could somehow be intercepted/mocked.
  */
 async function attachTimeInfo() {
-	// for a maximized video page -> hover over username
-	Array.from(document.querySelectorAll('span[data-e2e="browser-nickname"]')).map(span => {
-		if (attachedElements.has(span)) return; // skip already processed
+	// For a maximized video page -> hover over username
+	for (const span of Array.from(document.querySelectorAll('span[data-e2e="browser-nickname"]'))) {
```

Comment:
curiosity: are there strict rules against using .map in this context? I usually find it easier to read/code.

Reply:
So the change I did was `.map()` to `.forEach()`. `.map()` expects a return value and no side-effects, which engines optimize around.`.forEach()` is clearer for cases where you don't want to return anything, as is the case here.

The change to `for ... of ...` was then done by the linter.

Reply:
see https://github.com/microsoft/TypeScript/blob/main/src/lib/es5.d.ts#L1235-L1246